design/one-pager-release-process.md

```diff
@@ -0,0 +1,172 @@
+This pipeline would run the following build command:
+
+    ```bash
+    ./build/run make tag VERSION=${params.version}
+    ```
+
+* **promote**: This new pipeline would perform the promotion of a release build to a particular channel.
+We have currently been using `master` and `alpha` channels, and we can consider in the future changing to other channels such as `stable` and `edge`, but that is out of scope of this document.
+This pipeline would run the following build command:
+
+    ```bash
+    ./build/run make promote BRANCH_NAME=${BRANCH_NAME} VERSION=${params.version} CHANNEL=${params.channel}
+    ```
+
+## Stack Releases
+
+While this document has so far described the release process improvements for the [main Crossplane repository](https://github.com/crossplaneio/crossplane),
+there are also some infrastructure stacks that the Crossplane community currently owns and is responsible for releasing, such as [`stack-gcp`](https://github.com/crossplaneio/stack-gcp).
+Since these stack repositories currently also take advantage of the build submodule, they will also follow a very similar process as Crossplane and receive a similar set of improvements.
+
+Recall that a Stack is simply just an [OCI image](https://github.com/crossplaneio/crossplane/blob/master/design/design-doc-stacks.md#stack-package-format).
+Therefore, the build and publishing process can follow essentially the same process that Crossplane uses.
+Other functionality in the build submodule, such as creating a Helm chart, is not necessary for Stacks, but relevant functionality should definitely be reused.
+For example, building and publishing the stack package, tagging and creating a release branch, promoting the release, etc.
+
+The GCP, AWS, and Azure stacks will all receive new `tag` and `promote` pipelines similar to Crossplane, as well as receive a fix for [#330](https://github.com/crossplaneio/crossplane/issues/330).
+
+Note that each Stack can be released independently and on their own schedule, but for now they will follow the same monthly release schedule as Crossplane so the community can receive frequent updates.
+
+## Issues and Unknowns
+
+### Upgrade
+
+While Crossplane and Stacks do not have an officially supported upgrade path, this proposal for more frequent updates may cause some pain within the community.
+This isn't ideal but can be acceptable while we are still in alpha.
+We should prioritize working out our upgrade story to remove this obstacle and allow the community to take a hard dependency on running Crossplane across versions.
+
+### Commits after tagging
+
+Currently, our tagging logic performs the tag on the HEAD commit of `master` and then creates the release branch.
+This means that the very first build out of the release branch will be the official minor release, since you cannot re-tag in `git` without essentially rewriting history.
```

Comment:
Nitpick: I would describe the reason to not re-tag as more because it goes against what people expect out of the behavior of a tag, rather than because it is rewriting history. The standard expectation of a tag is that it is immutable, and git makes it hard to change a tag because of this, **not** because history is being rewritten. It used to be the case that `git fetch --tags` was sufficient to update changed tags, but it [looks like](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt---force) the command is now more like `git fetch --tags --force`. Anyway, there's some [additional discussion about expectations in the manpage for git tag](https://git-scm.com/docs/git-tag#_on_re_tagging).

Reply:
sounds good, i'll change the wording here to tags being immutable as opposed to re-writing history.